classes/migration/upgrade/OJSv3_3_0UpgradeMigration.inc.php

```diff
@@ -41,4 +43,121 @@ public function down() {
+						});
+					}
+				}
+			}
+		}
+
+		// Convert settings where only setting_type column is available
+		$tables = Capsule::connection()->getDoctrineSchemaManager()->listTableNames();
+		foreach ($tables as $tableName) {
+			if (substr($tableName, -9) !== '_settings' || in_array($tableName, $processedTables)) continue;
+			Capsule::table($tableName)->where('setting_type', 'object')->get()->each(function ($row) use ($tableName) {
+				$this->_toJSON($row, $tableName, ['setting_name', 'locale'], 'setting_value');
+			});
+		}
+
+		// Finally, convert values of other tables dependent from DAO::convertToDB
+		Capsule::table('review_form_responses')->where('response_type', 'object')->get()->each(function ($row) {
+			$this->_toJSON($row, 'review_form_responses', ['review_id'], 'response_value');
+		});
+
+		Capsule::table('site')->get()->each(function ($row) {
+			$oldInstalledLocales = unserialize($row->{'installed_locales'});
+			$oldSupportedLocales = unserialize($row->{'supported_locales'});
+
+			if (is_array($oldInstalledLocales) && $this->_isNumerical($oldInstalledLocales)) $oldInstalledLocales = array_values($oldInstalledLocales);
+			if (is_array($oldSupportedLocales) && $this->_isNumerical($oldSupportedLocales)) $oldSupportedLocales = array_values($oldSupportedLocales);
+
+			if ($oldInstalledLocales) {
+				$newInstalledLocales = json_encode($oldInstalledLocales, JSON_UNESCAPED_UNICODE);
+				Capsule::table('site')->take(1)->update(['installed_locales' => $newInstalledLocales]);
+			}
+
+			if ($oldSupportedLocales) {
+				$newSupportedLocales = json_encode($oldSupportedLocales, JSON_UNESCAPED_UNICODE);
+				Capsule::table('site')->take(1)->update(['supported_locales' => $newSupportedLocales]);
+			}
+		});
+	}
+
+	/**
+	 * @param $row stdClass row representation
+	 * @param $tableName string name of a settings table
+	 * @param $searchBy array additional parameters to the where clause that should be combined with AND operator
+	 * @param $valueToConvert string column name for values to convert to JSON
+	 * @return void
+	 */
+	private function _toJSON($row, $tableName, $searchBy, $valueToConvert)
+	{
+		$oldValue = unserialize($row->{$valueToConvert});
+		// Reset arrays to avoid keys being mixed up
+		if (is_array($oldValue) && $this->_isNumerical($oldValue)) $oldValue = array_values($oldValue);
+		if (!$oldValue && !is_array($oldValue)) return; // don't continue if value cannot be unserialized
+		$newValue = json_encode($oldValue, JSON_UNESCAPED_UNICODE); // don't convert utf-8 characters to unicode escaped code
+
+		$id = array_key_first((array)$row); // get first/primary key column
+
+		// Remove empty filters
+		$searchBy = array_filter($searchBy, function ($item) use ($row) {
+			if (empty($row->{$item})) return false;
+			return true;
+		});
+
+		// Build raw where query
+		$whereRaw = '';
+		$lastKey = array_key_last($searchBy);
+		foreach ($searchBy as $key => $column) {
```

Comment:
I'd suggest using the `where` function to construct the conditions rather than `whereRaw`; it's much easier to read (and won't involve string concatenation!)